sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java

```diff
@@ -273,165 +279,162 @@ public String getVaultUrl() {
     }
 
     /**
-     * Creates a {@link KeyVaultRoleAssignment} with a randomly generated {@link UUID name}.
+     * Creates a {@link KeyVaultRoleAssignment} with a randomly generated name.
      *
-     * @param roleScope The {@link KeyVaultRoleAssignmentScope roleScope} of the {@link KeyVaultRoleAssignment} to
+     * @param roleScope The {@link KeyVaultRoleScope roleScope} of the {@link KeyVaultRoleAssignment} to
      * create.
-     * @param properties Properties for the {@link KeyVaultRoleAssignment}.
+     * @param roleDefinitionId The {@link KeyVaultRoleDefinition role definition} ID for the role assignment.
+     * @param principalId The principal ID assigned to the role. This maps to the ID inside the Active Directory.
      * @return A {@link Mono} containing the created {@link KeyVaultRoleAssignment}.
-     * @throws NullPointerException if the {@link KeyVaultRoleAssignmentScope roleScope} or
-     * {@link KeyVaultRoleAssignmentProperties properties} are {@code null}.
+     * @throws NullPointerException if the {@link KeyVaultRoleScope roleScope}, {@link String roleAssignmentName},
+     * {@link String roleDefinitionId} or {@link String principalId} are {@code null}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<KeyVaultRoleAssignment> createRoleAssignment(KeyVaultRoleAssignmentScope roleScope, KeyVaultRoleAssignmentProperties properties) {
-        return createRoleAssignment(roleScope, UUID.randomUUID(), properties);
+    public Mono<KeyVaultRoleAssignment> createRoleAssignment(KeyVaultRoleScope roleScope, String roleDefinitionId, String principalId) {
+        return createRoleAssignment(roleScope, UUID.randomUUID().toString(), roleDefinitionId, principalId);
     }
 
     /**
      * Creates a {@link KeyVaultRoleAssignment}.
      *
-     * @param roleScope The {@link KeyVaultRoleAssignmentScope roleScope} of the {@link KeyVaultRoleAssignment} to
+     * @param roleScope The {@link KeyVaultRoleScope roleScope} of the {@link KeyVaultRoleAssignment} to
      * create.
-     * @param name The name used to create the {@link KeyVaultRoleAssignment}. It can be any valid UUID.
-     * @param properties Properties for the {@link KeyVaultRoleAssignment}.
+     * @param roleAssignmentName The roleAssignmentName used to create the {@link KeyVaultRoleAssignment}. It can be any
```

Comment:
should this be a `UUID`? I believe the service call will fail if it isn't a valid `UUID` string value.

Reply:
We decided to go with `String` and let the service validate to be consistent with other Java SDKs.

Reply:
Resolving this based on [this conversation](https://github.com/Azure/azure-sdk-for-java/pull/17284#discussion_r521692519).